.github/workflows/reusable_setup.yml

```
@@ -0,0 +1,22 @@
+name: "Setup"
+
+on: 
+  workflow_call:
+    secrets:
+      THE_GH_RELEASE_TOKEN:
+        required: false
+
+jobs:
+  setup:
+    strategy:
+      matrix:
+        os: ['ubuntu-22.04', 'macos-latest']
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Regarding the chances in 182e6322f4f2f070dc2a65dbf16db2e92da48947
 
 Even though the token isn't required, it should still be used if provided no? I believe something like what is provided below should work, using the action's default token if not provided.
 
 Same thing applies to the other `reusable(...)` actions

```suggestion
    steps:
      - uses: actions/checkout@v4
        with:
          fetch-depth: 0
          token: ${{ secrets.THE_GH_RELEASE_TOKEN || github.token }}
```